src/main/webapp/app/tasks/local-task.service.spec.ts

```diff
@@ -52,4 +53,21 @@ describe('LocalTaskService', () => {
     expect(localStorage.getItem).toHaveBeenCalled();
     expect(localStorage.setItem).toHaveBeenCalled();
   });
+
+  it('should update the status of the task', () => {
+    const data = {id: 'db1c4afe-d358-4b14-acfd-0953e882a6b6', name: 'Test Task', isCompleted: false};
+    // when
+    taskService.update(id, data);
+
+    // then
+    expect(data).toEqual({id: 'db1c4afe-d358-4b14-acfd-0953e882a6b6', name: 'Test Task', isCompleted: true});
+  });
+
+  it('should update the local storage with pending tasks', () => {
+    // when
+    taskService.clearCompletedTasks(mockTask);
+
+    // then
+    expect(localStorage.setItem).toHaveBeenCalled();
```

Comment:
you can test how the setItem is called with the method toHaveBeenCalledWith(). This would secure that you are deleting all tasks that completet

Reply:
Thanks, great input.